packages/bitcore-wallet-service/src/lib/emailservice.ts

```diff
@@ -326,7 +326,7 @@ export class EmailService {
       }
 
       if (_.includes(['NewIncomingTx', 'NewOutgoingTx'], notification.type) && data.txid) {
-        const urlTemplate = this.publicTxUrlTemplate[wallet.chain][wallet.network];
+        const urlTemplate = this.publicTxUrlTemplate[wallet.chain]?.[wallet.network];
```

Comment:
what about doing it this way ? 

`const chain = wallet.chain || ChainService.getChain(wallet.coin); // getChain -> backwards compatibility
  const urlTemplate = this.publicTxUrlTemplate[chain]?.[wallet.network];`

Reply:
most wallet don't have the `chain` property

Reply:
`this.storage.fetchWallet` already populates the chain field like that with `Wallet.fromObj`

Reply:
ooh ok nice..